### ISSUE TITLE
Mopro CLI

### ISSUE THREAD

**Issue by oskarth** [oskarth/mopro#77]
This is an umbrella issue for Mopro CLI.

# Problem

1) We want to make it easy for people to get started with Mopro
2) Developers want to work in their own repo
3) We want to minimize cognitive overhead and have one interface point, not a series of ad hoc scripts

## Details

![image](https://github.com/oskarth/mopro/assets/1552237/8dc22cfe-88cb-4f5b-a8a5-8c62a5b1aad7)

- Create a mopro CLI as the main way of interacting with mopro.
- As opposed to ad hoc scripts and instructions; should feel similar to something like Foundry
- init to initialize a project without having to work inside the mopro monorepo
  - Should create a new project with example code to run circuit on platforms
  - Options possible: --platform ios,android,pwa,desktop --language circom <MyProject> (not all have to be implemented)
  - Very rough start [here](https://github.com/oskarth/mopro/tree/main/mopro-cli) with Clap
- prepare to install dependencies
  - basically [prepare.sh](http://prepare.sh/) script - can shell out
- build to build all necessary dependencies
  - Options: --plaform ios,android,(react-native),(pwa)
- update to update bindings
- test to run tests
- More options might be added; basically a minimal surface area that makes mopro users productive quickly

Initial work started here https://github.com/oskarth/mopro/tree/main/mopro-cli

## Acceptance criteria

- Mopro CLI is the main way of interfacing with Mopro, not ad hoc scripts
- Developers can create a project outside of their repo and work there
- It is in an MVP stage where it can be used end-to-end

**Comment by oskarth** [comment]
- MVP https://github.com/oskarth/mopro/issues/78
- iOS support https://github.com/oskarth/mopro/issues/92
- Enhancements https://github.com/oskarth/mopro/issues/93

**Comment by oskarth** [comment]
Done
